implants/golem/src/main.rs

```diff
@@ -99,34 +122,65 @@ fn main() -> anyhow::Result<()> {
+        // Handle embedded tomes (potentially encrypted)
         let mut parsed_tomes: Vec<ParsedTome> = Vec::new();
-        for embedded_file_path in eldritch::assets::Asset::iter() {
-            let filename = embedded_file_path.split('/').last().unwrap_or("");
-            println!("{}", embedded_file_path);
-            if filename == "main.eldritch" {
-                let tome_contents_extraction_result =
-                    match eldritch::assets::Asset::get(embedded_file_path.as_ref()) {
-                        Some(local_tome_content) => {
-                            String::from_utf8(local_tome_content.data.to_vec())
-                        }
-                        None => {
-                            eprint!("Failed to extract eldritch script as string");
-                            Ok("".to_string())
-                        }
-                    };
 
-                let tome_contents = match tome_contents_extraction_result {
-                    Ok(local_tome_contents) => local_tome_contents,
-                    Err(utf8_error) => {
-                        eprint!("Failed to extract eldritch script as string {utf8_error}");
-                        "".to_string()
+        let password = match matches.get_one::<String>("password") {
+            Some(p) => p,
+            None => {
+                eprintln!("Password required for encrypted embedded tomes. Use --password <PASSWORD> or -p <PASSWORD>");
+                process::exit(1);
+            }
+        };
+
+        // Derive candidate AES key from CLI password and compile-time KDF_SALT
+        let mut candidate_aes_key = [0u8; 32]; // AES-256 key size
+        pbkdf2_hmac::<Sha256>(
+            password.as_bytes(),
+            KDF_SALT.as_bytes(),
+            10_000, // Iterations, must match build.rs
+            &mut candidate_aes_key,
+        );
+
+        // Verify password by comparing derived key with compile-time AES_KEY
+        if candidate_aes_key != *AES_KEY {
+            eprintln!("Invalid password.");
+            process::exit(1);
+        }
+
+        // Password is valid, proceed to decrypt and load tomes
+        let cipher = Aes256Gcm::new(Key::<Aes256Gcm>::from_slice(AES_KEY));
+
+        for file in GOLEM_ENCRYPTED_TOMES.files() {
+            if file.path().file_name().map_or(false, |name| name == "main.eldritch") {
+                let encrypted_content = file.contents();
+                let nonce = Nonce::from_slice(AES_NONCE);
+                match cipher.decrypt(nonce, encrypted_content) {
+                    Ok(decrypted_data) => {
+                        match String::from_utf8(decrypted_data) {
+                            Ok(tome_contents) => {
+                                parsed_tomes.push(ParsedTome { eldritch: tome_contents });
+                            }
+                            Err(e) => {
+                                eprintln!("Failed to convert decrypted tome to UTF-8 string: {}. Skipping.", e);
```

Comment:
I'm not sure this is the right way to print errors.
Double check how error printing is handled in other places.
Usually we need a cfg_attr(debug) or if it's an error passed up through grpc we use report.error()